### PR TITLE
tqdm imports now supporting console and notebook modes

### PR DESCRIPTION
**Pull request by vnherdeiro**
Prototyping a pymc3 model inside a Jupyter notebook the tqdm progress bars can get ugly or buggy. There is a version of these progress bars tailored for notebooks and addressing these issues. The imports are now aware of the environment and load the appropriate ones.

### CODE REVIEW

File: pymc3/parallel_sampling.py [hunk 1]
```diff
@@ -347,9 +347,7 @@ def __init__(
         progressbar=True,
     ):
         if progressbar:
-            import tqdm
-
-            tqdm_ = tqdm.tqdm
+            from tqdm.auto import tqdm as tqdm_
```
Comment (twiecki) [inline pymc3/parallel_sampling.py:350]: We can probably just change `tqdm_` below to `tqdm`, the existing pattern here is pretty weird.

**Comment by twiecki** [comment]
Thanks - this looks great! Also needs a line in the release-notes.

**Comment by rpgoldman** [comment]
What about squashing this together into one commit before merging?  In particular, we don't really need "fixed the `tqdm.trange` calls" in the history...

Does `requirements-dev` also need updating?

**Comment by twiecki** [comment]
@rpgoldman We can just squash and merge on github now.

**Comment by rpgoldman** [comment]
> @rpgoldman We can just squash and merge on github now.

Sounds good to go then.

I see that `requirements-dev` *complements* `requirements`, so it does not need an update.

Thanks, @vnherdeiro !

**Comment by vnherdeiro** [comment]
I am gonna clean the working tree and do a new pull request. A bit brute force, but effective.

edit: done.

**Comment by vnherdeiro** [comment]
> Does `requirements-dev` also need updating?

I do not think so.

**Comment by rpgoldman** [comment]
@vnherdeiro 
> > Does `requirements-dev` also need updating?
> 
> I do not think so.

You were right about this; I was wrong -- the developer requirements are _added to_ the requirements, so the development environment will be correct.

**Pull request closed without merging**